components/camel-debezium/src/main/java/org/apache/camel/component/debezium/DebeziumEndpoint.java

```diff
@@ -0,0 +1,133 @@
+
+/**
+ * Represents a Debezium endpoint which is used for interacting with Debezium
+ * embedded engine.
+ */
+@UriEndpoint(firstVersion = "3.0.0", scheme = "debezium", title = "Debezium", syntax = "debezium:connector-name", label = "cdc", consumerOnly = true)
+public class DebeziumEndpoint extends DefaultEndpoint {
+
+    @UriParam
+    private EmbeddedDebeziumConfiguration configuration;
+
+    // is only meant to include different configurations for each connector that is generated by camel-maven plugin
+    @UriParam
+    private MySqlConnectorEmbeddedDebeziumConfiguration mySqlConnectorEmbeddedDebeziumConfiguration;
+
+    public DebeziumEndpoint(String uri, DebeziumComponent component, EmbeddedDebeziumConfiguration config) {
+        super(uri, component);
+        this.configuration = config;
+    }
+
+    public DebeziumEndpoint() {
+    }
+
+    @Override
+    public Producer createProducer() throws Exception {
+        throw new UnsupportedOperationException("Cannot produce from a DebeziumEndpoint: "
+                                                + getEndpointUri());
+    }
+
+    public Consumer createConsumer(Processor processor) throws Exception {
+        return new DebeziumConsumer(this, processor);
+    }
+
+    public boolean isSingleton() {
+        return true;
+    }
+
+    public EmbeddedDebeziumConfiguration getConfiguration() {
+        return configuration;
+    }
+
+    public void setConfiguration(EmbeddedDebeziumConfiguration configuration) {
+        this.configuration = configuration;
+    }
+
+    public MySqlConnectorEmbeddedDebeziumConfiguration getMySqlConnectorEmbeddedDebeziumConfiguration() {
+        return mySqlConnectorEmbeddedDebeziumConfiguration;
+    }
+
+    public ExecutorService createExecutor() {
+        return getCamelContext().getExecutorServiceManager().newSingleThreadExecutor(this,
+                                                                                     "DebeziumConsumer");
+    }
+
+    public Exchange createDbzExchange(final SourceRecord record) {
+        final Exchange exchange = super.createExchange();
+
+        final Message message = exchange.getIn();
+
+        message.setHeader(DebeziumConstants.FIELD_IDENTIFIER, record.topic());
+        message.setHeader(DebeziumConstants.FIELD_KEY, record.key());
+        message.setHeader(DebeziumConstants.FIELD_SOURCE, record.sourceOffset());
+        setOperationFieldInHeader(message, record.valueSchema(), record.value());
+
+        message.setBody(record.value());
+
+        return exchange;
+    }
+
+    private void setOperationFieldInHeader(final Message message, final Schema schema, final Object value) {
+        if (isValueAndStructSchemaSet(schema, value)) {
+            message.setHeader(DebeziumConstants.FIELD_OPERATION, getOperationFromValueRecord((Struct)value));
+        } else if (!isValueAndSchemaSet(schema, value)) { // sometimes a delete will be emitted as null in the
```

Comment:
Hey @gunnarmorling, I am bit hesitant here, do I need as well to check for a `null` value for the delete? I noticed that DBZ emits two messages sequentially when a delete operation occurs, the first with `op=d` and the second with a {{null}} value, I guess would be enough to relay on the `op` and ignore the second?

Reply:
It's a configuration of the Debezium connectors whether to omit that second message ("tombstone") or not. It's used as the indicator to the Kafka broker that all the messages with the same key can be removed from the topic during compaction. It probably doesn't not make that much sense for the Camel use case, so you might default that option to not emitting the tombstone. I think though the Camel connector should still accept it gracefully if it is sent.

Reply:
Indeed, I will keep it as it is to let camel accept it and default that option to not use tombstones